lib/chef/provider/powershell_script.rb

```diff
@@ -129,7 +192,80 @@ def validate_script_syntax!
       # last process run in the script if it is the last command
       # executed, otherwise 0 or 1 based on whether $? is set to true
       # (success, where we return 0) or false (where we return 1).
-      def wrapper_script
+      def inline_powershell_wrapper_script
```

Comment:
Thinking this can potentially be deduplicated by some sort of templating or builder logic so that the behavior stays consistent between them.  cc: @dafyddcrosby

Reply:
yeah, i think that's sane to do - i'll make that happen

Reply:
In the latest push, this is simplified to a single function with the two changes as simple conditions at those line points, which seemed much easier to read to my eye.